tests/phpunit/tests/option/networkOption.php

```diff
@@ -689,14 +689,122 @@ public function data_stored_as_empty_string() {
 		);
 	}
 
+	/**
+	 * Test cases for testing whether update_network_option() will add a non-existent option.
+	 */
+	public function data_option_values() {
+		return array(
+			array( '1' ),
+			array( 1 ),
+			array( 1.0 ),
+			array( true ),
+			array( 'true' ),
+			array( '0' ),
+			array( 0 ),
+			array( 0.0 ),
+			array( false ),
+			array( '' ),
+			array( null ),
+			array( array() ),
+		);
+	}
+
+	/**
+	 * Tests that a non-existent option is added only when the pre-filter matches the default 'false'.
+	 *
+	 * @ticket 59360
+	 * @dataProvider data_option_values
+	 *
+	 * @covers ::update_option
+	 */
+	public function test_update_option_with_false_pre_filter_adds_missing_option( $option ) {
+		// Filter the old option value to `false`.
+		add_filter( 'pre_option_foo', '__return_false' );
+		add_filter( 'pre_site_option_foo', '__return_false' );
+
+		/*
+		 * When the network option is equal to the filtered version, update option will bail early.
+		 * Otherwise, The pre-filter will make the old option `false`, which is equal to the
+		 * default value. This causes an add_network_option() to be triggered.
+		 */
```

Comment:
Not sure this comment is accurate, as this was copy/pasta from the update_option tests.